src/Native/System.Native/pal_networking.cpp

```diff
@@ -255,10 +245,12 @@ SystemNative_IPv6StringToAddress(const uint8_t* address, const uint8_t* port, ui
     assert(buffer != nullptr);
     assert(bufferLength == NUM_BYTES_IN_IPV6_ADDRESS);
     assert(scope != nullptr);
+    assert(address != nullptr);
+
+    addrinfo hint = { .ai_family = AF_INET6, .ai_flags = AI_NUMERICHOST | AI_NUMERICSERV };
 
-    // Call our helper to do the getaddrinfo call for us; once we have the info, copy what we need
-    addrinfo* info;
-    int32_t result = IpStringToAddressHelper(address, port, true, info);
+    addrinfo* info = nullptr;
+    int32_t result = getaddrinfo(reinterpret_cast<const char*>(address), reinterpret_cast<const char*>(port), &hint, &info);
```

Comment:
Should we just make the function's parameters be ```const char*``` instead of ```const uint8_t*```?  Shouldn't be a breaking change, and it seems like by using uint8_t instead of char we're fighting a battle against the APIs we're then trying to call.  Not a big deal, though.